File: src/main/java/com/fish/extendedae_plus_client/impl/cache/CacheCrafting.java
package com.fish.extendedae_plus_client.impl.cache;

public final class CacheCrafting {
    private static int markedPlan = 0;
    private static boolean opening;

    public static void markPlan() {
        markedPlan++;
    }

    public static void cancelPlan() {
        markedPlan--;
    }

    public static boolean isEmpty() {
        return markedPlan == 0;
    }

    public static boolean isOpening() {
        if (opening) {
            opening = false;
            return true;
        } else return false;
    }

    public static void setOpening(boolean opening) {
        CacheCrafting.opening = opening;
    }
}

File: src/main/kotlin/com/fish/extendedae_plus_client/impl/cache/CacheCrafting.kt
package com.fish.extendedae_plus_client.impl.cache

import kotlin.math.max

object CacheCrafting {
    private var markedPlan = 0

    @JvmStatic
    var isOpening: Boolean = false
        get() {
            if (field) {
                field = false
                return true
            } else return false
        }

    @JvmStatic
    val isEmpty: Boolean
        get() = markedPlan == 0

    @JvmStatic
    fun markPlan() {
        markedPlan++
    }

    @JvmStatic
    fun cancelPlan() {
        markedPlan = max(markedPlan - 1, 0)
    }

    fun clear() {
        this.isOpening = false
        this.markedPlan = 0
    }
}
